lib/fluent/command/cat.rb

```diff
@@ -20,320 +20,347 @@
+        $stdout.puts "!#{time}:#{Yajl.dump(record)}"
       end
-    }
-  end
+    end
 
-  def close
-    @socket.close
-    @socket = nil
-  end
+    def configure_option_parser
+      @opt_praser = OptionParser.new
 
-  def start
-    @timer = TimerThread.new(self)
-    @timer.start
-    self
-  end
+      @opt_parser.banner += " <tag>"
+      @opt_parser.version = Fluent::VERSION
 
-  def shutdown
-    @timer.shutdown
-  end
+      @port = 24224
+      @host = '127.0.0.1'
+      @unix = false
+      @socket_path = Fluent::DEFAULT_SOCKET_PATH
 
-  private
-  def write_impl(array)
-    socket = get_socket
-    unless socket
-      return false
-    end
+      @config_path = Fluent::DEFAULT_CONFIG_PATH
+      @format = 'json'
+      @message_key = 'message'
+      @time_as_integer = false
+      @retry_limit = 5
 
-    begin
-      packer = Fluent::MessagePackFactory.packer
-      socket.write packer.pack([@tag, array])
-      socket.flush
-    rescue
-      $stderr.puts "write failed: #{$!}"
-      close
-      return false
-    end
+      @opt_parser.on('-p', '--port PORT', "fluent tcp port (default: #{@port})", Integer) {|i|
+        @port = i
+      }
 
-    return true
-  end
+      @opt_parser.on('-h', '--host HOST', "fluent host (default: #{@host})") {|s|
+        @host = s
+      }
 
-  def get_socket
-    unless @socket
-      unless try_connect
-        return nil
-      end
-    end
+      @opt_parser.on('-u', '--unix', "use unix socket instead of tcp", TrueClass) {|b|
+        @unix = b
+      }
 
-    @socket_time = Time.now.to_i
-    return @socket
-  end
+      @opt_parser.on('-s', '--socket PATH', "unix socket path (default: #{@socket_path})") {|s|
+        @socket_path = s
+      }
+
+      @opt_parser.on('-f', '--format FORMAT', "input format (default: #{@format})") {|s|
+        @format = s
+      }
+
+      @opt_parser.on('--json', "same as: -f json", TrueClass) {|b|
+        @format = 'json'
+      }
+
+      @opt_parser.on('--msgpack', "same as: -f msgpack", TrueClass) {|b|
+        @format = 'msgpack'
+      }
 
-  def try_connect
-    begin
-      now = Time.now.to_i
+      @opt_parser.on('--none', "same as: -f none", TrueClass) {|b|
+        @format = 'none'
+      }
 
-      unless @error_history.empty?
-        # wait before re-connecting
-        wait = 1 #@retry_wait * (2 ** (@error_history.size-1))
-        if now <= @socket_time + wait
-          sleep(wait)
-          try_connect
+      @opt_parser.on('--message-key KEY', "key field for none format (default: #{@message_key})") {|s|
+        @message_key = s
+      }
+
+      @opt_parser.on('--time-as-integer', "Send time as integer for v0.12 or earlier", TrueClass) { |b|
+        time_as_integer = true
+      }
+
+      @opt_parser.on('--retry-limit N', "Specify the number of retry limit (default: #{@retry_limit})", Integer) {|n|
+        @retry_limit = n
+      }
+
+      singleton_class.module_eval do
+        define_method(:usage) do |msg|
```

Comment:
It should be an instance method since it's not global scope anymore.

BTW I couldn't understand that what the original code dose until I read the following article: https://q.hatena.ne.jp/1376406562
We shouldn't leave such hacky code.